src/Plugins/ITKImageProcessing/src/ITKImageProcessing/Filters/ITKMhaFileReader.cpp

```diff
@@ -0,0 +1,441 @@
+    ITKImageReader imageReaderFilter;
+    Arguments args;
+    args.insertOrAssign(ITKImageReader::k_FileName_Key, fileNamePath);
+    args.insertOrAssign(ITKImageReader::k_ImageGeometryPath_Key, imageGeomPath);
+    args.insertOrAssign(ITKImageReader::k_CellDataName_Key, cellDataName);
+    args.insertOrAssign(ITKImageReader::k_ImageDataArrayPath_Key, imageDataArrayPath);
+
+    auto executeResult = imageReaderFilter.execute(importedDataStructure, args, pipelineNode, messageHandler, shouldCancel);
+    if(executeResult.result.invalid())
+    {
+      return executeResult.result;
+    }
+
+    Result<> copyResult;
+    const auto& srcDataArray = importedDataStructure.getDataRefAs<IDataArray>(imageDataArrayPath);
+    const auto& destDataArray = dataStructure.getDataRefAs<IDataArray>(imageDataArrayPath);
+    messageHandler(fmt::format("Copying image data to destination array '{}'...", destDataArray.getName()));
+    switch(srcDataArray.getDataType())
+    {
+    case DataType::uint8: {
+      copyResult = CopyImageData<uint8>(importedDataStructure, dataStructure, imageDataArrayPath, fileNamePath);
+      break;
+    }
+    case DataType::int8: {
+      copyResult = CopyImageData<int8>(importedDataStructure, dataStructure, imageDataArrayPath, fileNamePath);
+      break;
+    }
+    case DataType::uint16: {
+      copyResult = CopyImageData<uint16>(importedDataStructure, dataStructure, imageDataArrayPath, fileNamePath);
+      break;
+    }
+    case DataType::int16: {
+      copyResult = CopyImageData<int16>(importedDataStructure, dataStructure, imageDataArrayPath, fileNamePath);
+      break;
+    }
+    case DataType::uint32: {
+      copyResult = CopyImageData<uint32>(importedDataStructure, dataStructure, imageDataArrayPath, fileNamePath);
+      break;
+    }
+    case DataType::int32: {
+      copyResult = CopyImageData<int32>(importedDataStructure, dataStructure, imageDataArrayPath, fileNamePath);
+      break;
+    }
+    case DataType::uint64: {
+      copyResult = CopyImageData<uint64>(importedDataStructure, dataStructure, imageDataArrayPath, fileNamePath);
+      break;
+    }
+    case DataType::int64: {
+      copyResult = CopyImageData<int64>(importedDataStructure, dataStructure, imageDataArrayPath, fileNamePath);
+      break;
+    }
+    case DataType::float32: {
+      copyResult = CopyImageData<float32>(importedDataStructure, dataStructure, imageDataArrayPath, fileNamePath);
+      break;
+    }
+    case DataType::float64: {
+      copyResult = CopyImageData<float64>(importedDataStructure, dataStructure, imageDataArrayPath, fileNamePath);
+      break;
+    }
+    case DataType::boolean: {
+      copyResult = CopyImageData<bool>(importedDataStructure, dataStructure, imageDataArrayPath, fileNamePath);
+      break;
+    }
+    default: {
+      return MakeErrorResult(-5006, fmt::format("The created data array '{}' does not have a compatible DataType.", srcDataArray.getName()));
+    }
+    }
```

Comment:
```suggestion
copyResult = ExecuteDataFunction(CopyImageDataFunctor{}, srcDataArray.getDataType(), importedDataStructure, dataStructure, imageDataArrayPath, fileNamePath);
if (copyResult.invalid())
{
  return copyResult;
}
```
You'll have to double check my syntax for the two updates I proposed here since I didn't double check them myself and you'll have to add in the include for the FilterUtilities.hpp file but I believe you should be able to simply to this.